### PR TITLE
Add setuptools integration.

### PR DESCRIPTION
**Pull request by dave-shawley**
This PR adds a setuptools command for running `safety check`.  I'm not sure if this is something that you are interested in or not, but it makes configuration via *setup.cfg* possible.

### CODE REVIEW

**Comment by 3clypse** [comment]
It's a great feature to add :)

File: setup.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 ]
 
 test_requirements = [
-    # TODO: put package test requirements here
+    'mock==1.0.1',
```
[the remark this answers is not in the document]
Comment (dave-shawley) [reply setup.py]: Sounds good ... I rebased against master and update tox.ini appropriately.

File: README.md [hunk 2]
```diff
@@ -147,6 +147,22 @@ and displays a status on GitHub.
...
+
+```
+python setup.py safety --full-report
+```
+
+The command-line options can also be configured in *setup.cfg*:
+
+```
+[safety]
+full-report yes
```
[the remark this answers is not in the document]
Comment (dave-shawley) [reply README.md]: Yes ... yes you do 😞

**Comment by dave-shawley** [comment]
> My understanding is that the setup command, as another interface to Safety, should be adapt itself to such context. So, instead of allowing users to pass any argument as they would to to Safety CLI, just makes it sound like having two ways of doing the same thing. What I am trying to say is that users could then just call the CLI and that's it.
> 
> So, my proposal here is to change the arguments interface, and how the command gathers options from `setup` call itself, to run the check.
> 
> I will give you an example:
> 
> 1. Remove all options except for those changing results format and the key: bare, json, full-report and key
> 2. Bring minimum set of options to `setup` call like: ignore, output, cache, db and files
> 3. Make sure we are running a check against `install_requires` dependencies by default.
> 
> In a way or another, this already looks great! Looking forward your changes.

Using `install_requires` is a perfect default... not sure why I didn't think of that.  The main reason that I offered all of the options is to make them configurable via the _setup.cfg_.  It does make sense to remove some of them.

1. Remove support for the proxy parameters
2. Remove support for the `--stdin` parameter
3. Initialize the list of packages with `install_requires`
4. Add packages explicitly specified in the `files` parameter to the list of packages

Does that sound good?

**Comment by rafaelpivato** [comment]
That would be great, @dave-shawley 

Looking forward those changes!

**Comment by Lucas-C** [comment]
Hi!
This looks very useful :)
What's the status of this PR?

**Pull request closed without merging**